Internals/Common/Utilities/TimeUtils.cs

```diff
@@ -1,64 +1,155 @@
+
+        if (span.Days != 0)
+            sb.Append(span.Days).Append(between);
+        if (span.Hours != 0)
+            sb.Append(span.Hours).Append(between);
+        if (span.Minutes != 0)
+            sb.Append(span.Minutes).Append(between);
+        if (span.Seconds != 0)
+            sb.Append(span.Seconds).Append(between);
+
+        return sb.ToString();
     }
 
-    public static string StringFormatCustom(this DateTime span, string between)
-    {
-        string days = span.Day == 0 ? "" : (span.Day == 1 ? $"{span.Day}{between}" : $"{span.Day}{between}");
-        string hours = span.Hour == 0 ? "" : (span.Hour == 1 ? $"{span.Hour}{between}" : $"{span.Hour}{between}");
-        string mins = span.Minute == 0 ? "" : (span.Minute == 1 ? $"{span.Minute}{between}" : $"{span.Minute}{between}");
-        string secs = span.Second == 0 ? "" : (span.Second == 1 ? $"{span.Second}" : $"{span.Second}");
+    public static string StringFormatCustom(this DateTime span, string between) {
+        StringBuilder sb = new(2 * 4 + between.Length * 4);
+
+        if (span.Day != 0)
+            sb.Append(span.Day).Append(between);
+        if (span.Hour != 0)
+            sb.Append(span.Hour).Append(between);
+        if (span.Minute != 0)
+            sb.Append(span.Minute).Append(between);
+        if (span.Second != 0)
+            sb.Append(span.Second).Append(between);
 
-        return $"{days}{hours}{mins}{secs}";
+        return sb.ToString();
     }
 
-    public static string StopwatchFormat(this TimeSpan span)
-    {
-        string hours;
-        if (span.Hours >= 10 && span.Hours < 100)
-            hours = $"0{span.Hours}";
-        else if (span.Hours < 10)
-            hours = $"00{span.Hours}";
+    public static string StopwatchFormat(this Stopwatch watch) {
+        var span = watch.Elapsed;
+        StringBuilder sb = new(64);
+        
+        // Append Hours
+        switch (span.Hours) {
+            case >= 10 and < 100:
+                sb.Append('0').Append(span.Hours);
+                break;
+            case < 10:
+                sb.Append("00").Append(span.Hours);
+                break;
+            default:
+                sb.Append(span.Hours);
+                break;
+        }
+
+        // Appends Minutes
+        if (span.Minutes < 10)
+            sb.Append('0').Append(span.Minutes);
+        else
+            sb.Append(span.Minutes);
+        
+        // Append Seconds
+        if (span.Seconds < 10)
+            sb.Append('0').Append(span.Seconds);
         else
-            hours = $"{span.Hours}";
-        string mins = span.Minutes < 10 ? $"0{span.Minutes}" : $"{span.Minutes}";
-        string secs = span.Seconds < 10 ? $"0{span.Seconds}" : $"{span.Seconds}";
-        int millisecs = span.Milliseconds;
+            sb.Append(span.Seconds);
+
+        // Append ms
+        sb.Append(span.Milliseconds);
 
-        return $"{hours}:{mins}:{secs}:{millisecs}";
+        return sb.ToString();
```

Comment:
Not sure how I feel about the changes to `StringBuilder`s. If you can explain why you chose this, I'll consider keeping it this way.